tests/rsa_tests.rs

```diff
@@ -163,26 +185,27 @@ fn test_signature_rsa_pkcs1_verify() {
             // Sanity check that we correctly DER-encoded the originally-
             // provided separate (n, e) components. When we add test vectors
             // for improperly-encoded signatures, we'll have to revisit this.
-            assert!(untrusted::Input::from(&public_key)
-                .read_all(error::Unspecified, |input| der::nested(
-                    input,
-                    der::Tag::Sequence,
-                    error::Unspecified,
-                    |input| {
+            let key_bits = untrusted::Input::from(&public_key)
+                .read_all(error::Unspecified, |input| {
+                    der::nested(input, der::Tag::Sequence, error::Unspecified, |input| {
+                        let n_len = der::positive_integer(input)?
+                            .big_endian_without_leading_zero()
+                            .len();
                         let _ = der::positive_integer(input)?;
-                        let _ = der::positive_integer(input)?;
-                        Ok(())
-                    }
-                ))
-                .is_ok());
+                        Ok(n_len * 8)
+                    })
+                })
+                .expect("invalid DER");
 
             let msg = test_case.consume_bytes("Msg");
             let sig = test_case.consume_bytes("Sig");
             let is_valid = test_case.consume_string("Result") == "P";
-
-            let actual_result =
-                signature::UnparsedPublicKey::new(alg, &public_key).verify(&msg, &sig);
-            assert_eq!(actual_result.is_ok(), is_valid);
+            for &(alg, min_bits) in params {
+                let width_ok = key_bits >= min_bits;
```

Comment:
In particular, we should have test cases for public keys of lengths 1016, 1017, 1018, 1019, 1020, 1021, 1022, 1023, 1024, etc., `width_ok` would be different than what this test would indicate, right?